build/Targets/Imports.targets

```diff
@@ -353,6 +369,7 @@
 
   <Import Project="GenerateAssemblyInfo.targets" Condition="'$(ProjectLanguage)' == 'CSharp' OR '$(ProjectLanguage)' == 'VB'" />
   <Import Project="GenerateInternalsVisibleTo.targets" />
+  <Import Project="DisableTransitiveReferences.targets" Condition="'$(RoslynProjectType)' == 'Vsix' AND '$(RoslynSdkProject)' == 'true'" />
```

Comment:
The commit which added this contains all of the details on why

https://github.com/dotnet/roslyn/pull/20333/commits/4d69e674335f463210cb3054b03e5b2b0967b91d

This is true for virtually every change here. Tried to keep them small and well documented to the reasons.